nilearn/_utils/data_gen.py

```diff
@@ -530,67 +756,97 @@ def basic_paradigm(condition_names_have_spaces=False):
     return events
 
 
-def basic_confounds(length):
+def basic_confounds(length, random_state=0):
+    """Generate basic confounds
+
+    Parameters
+    ----------
+    length : :obj:`int`
+        Length of basic confounds.
+
+    random_state : :obj:`int` or :obj:`numpy.random.RandomState` instance, \
+                   optional
+        Random number generator, or seed.
+        Default=0.
+
+    Returns
+    -------
+    confounds : :obj:`pandas.DataFrame`.
+        Basic confounds.
+
+    """
+    rand_gen = check_random_state(random_state)
     columns = ['RotX', 'RotY', 'RotZ', 'X', 'Y', 'Z']
-    data = np.random.rand(length, 6)
+    data = rand_gen.rand(length, 6)
     confounds = pd.DataFrame(data, columns=columns)
     return confounds
 
 
-def create_fake_bids_dataset(base_dir='', n_sub=10, n_ses=2,
+def create_fake_bids_dataset(base_dir='',
+                             n_sub=10,
+                             n_ses=2,
                              tasks=['localizer', 'main'],
-                             n_runs=[1, 3], with_derivatives=True,
+                             n_runs=[1, 3],
+                             with_derivatives=True,
                              with_confounds=True,
                              confounds_tag="desc-confounds_timeseries",
-                             no_session=False):
+                             no_session=False,
+                             random_state=0):
     """Creates a fake :term:`bids<BIDS>` dataset directory with dummy files.
     Returns fake dataset directory name.
```

Comment:
```suggestion
    """Creates a fake :term:`bids<BIDS>` dataset directory with dummy files.

    Returns fake dataset directory name.
```

I don't think we use it at the moment, but `flake8-docstrings` requires the summary sentence to use one line, and then the longer description should be after an empty line. It might be good to use that organization generally.

Reply:
Could you also mention somewhere in the docstring that all runs will be created with 100 volumes?

Reply:
Yes, done